Sources/Core/Shared/ExclusivityManager.swift

```diff
@@ -24,41 +24,48 @@ internal class ExclusivityManager {
         return dispatch_sync(queue) { self._addOperation(operation, category: category) }
     }
 
-    func removeOperation(operation: Operation, category: String) {
-        dispatch_async(queue) {
-            self._removeOperation(operation, category: category)
-        }
-    }
-
     private func _addOperation(operation: Operation, category: String) -> NSOperation? {
-        operation.log.verbose(">>> \(category)")
 
-        operation.addObserver(DidFinishObserver { [unowned self] op, _ in
-            self.removeOperation(op, category: category)
-        })
-
-        var operationsWithThisCategory = operations[category] ?? []
+        let operationsWithThisCategory = operations[category] ?? []
 
         let previous = operationsWithThisCategory.last
 
         if let previous = previous {
```

Comment:
Because `operationsWithThisCategory` is only modified when an operation is executed, if I create a bunch of operations, add MutuallyExclusive conditions to them, and then add them all to a queue, none of them will be dependent on each other with the current code (i.e. `operationsWithThisCategory.last` will always be `nil`).